exasol/bucketfs/__init__.py

```diff
@@ -55,14 +56,24 @@
     as_hash,
     as_string,
 )
-from exasol.bucketfs._error import BucketFsError
+from exasol.bucketfs._path import (
+    PathLike,
+    SystemType,
+    build_path
+)
+from exasol.bucketfs._error import BucketFsError, MountedBucketFsError
 from exasol.bucketfs._service import Service
 
 __all__ = [
     "Service",
+    "BucketLike",
     "Bucket",
     "MappedBucket",
     "BucketFsError",
+    "MountedBucketFsError",
+    "PathLike",
+    "SystemType",
+    "build_path",
```

Comment:
I think for all the path related stuff we should expose the `path` package/module.

example usage:
```python
from exasol import bucketfs

path = bucketfs.path.build_path(...)
```